api/base/serializers.py

```diff
@@ -1213,6 +1213,36 @@ def _handle_callable_view(self, obj, view):
         return view(getattr(obj, self.view_lambda_argument))
 
 
+class GenericRelationshipField(RelationshipField):
```

Comment:
Should we just be re-using the `TargetField` that Files use to resolve their GenericForeignKey? https://github.com/CenterForOpenScience/osf.io/blob/2ca525a761d2b27cc7f244226ea55166fa4ea9b2/api/base/serializers.py#L936

Reply:
The TargetField relies on the resource having a guid referent, I'm actually thinking of removing all these TargetField stuff, because it's a big class and it's only used 2 places and the `GenericRelationshipField` is move explicit and flexible. Any thoughts? Honestly it looks like a lot of these fields could be replaced with GenericRelationshipFields.